Lib/test/test_source_encoding.py

```diff
@@ -226,5 +226,18 @@ def check_script_output(self, src, expected):
         self.assertEqual(res.out.rstrip(), expected)
 
 
+class EncodingsTest(unittest.TestCase):
+
+    def test_bpo39337(self):
+        """
+        bpo-39337: similar to _Py_normalize_encoding(),
+        encodings.normalize_encoding() should ignore non-ASCII letters .
+        """
+        import encodings
+
+        out = encodings.normalize_encoding("���-8")
```

Comment:
Rather than using the same non-ASCII character 3 times, I suggest my favorite trio (latin1, UCS-2, UCS-4 string types):

```
"\xE9\u20AC\U0010ffff"
```

I suggest to test that `"utf\xE9\u20AC\U0010ffff-8"` gives`"utf_8"`.

Reply:
> Rather than using the same non-ASCII character 3 times

Oh, I use the different non-ASCII character at the first begining. Looks this file transfer my characters.

```
Lib/test/test_source_encoding.py: Python script, ISO-8859 text executable
```

> I suggest to test that "utf\xE9\u20AC\U0010ffff-8" gives"utf_8".

OK, let me try it.

Reply:
> Looks this file transfer my characters.

Well, I suggest to use \xHH, \uHHHH and \UHHHHHHHH syntax to avoid such bad surprises, and help other developers with similar encoding issue to maintain/review this test ;-) Sadly in 2020, ASCII-only is still the safest option to avoid any encoding or rendering issue.